phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -207,10 +210,120 @@ public void testSkipPostIndexPartialRowUpdate() throws Exception {
+            ResultSet rs = conn.createStatement().executeQuery(selectSql);
+            assertTrue(rs.next());
+            assertEquals("abc", rs.getString(1));
+            assertEquals("abcd", rs.getString(2));
+            assertFalse(rs.next());
+            selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val2  = 'abcc'";
+            // Verify that we will read from the second index table
+            assertExplainPlan(conn, selectSql, dataTableName, indexTableName + "2");
+            rs = conn.createStatement().executeQuery(selectSql);
+            // Verify that one phase writes have no effect
+            assertFalse(rs.next());
+            // Configure IndexRegionObserver to skip the last two write phase (i.e., the data table update and post index
+            // update phase) and check that this does not impact the correctness  (two overwrites)
+            IndexRegionObserver.setSkipDataTableUpdatesForTesting(true);
+            IndexRegionObserver.setSkipPostIndexUpdatesForTesting(true);
+            conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abccc')");
+            conn.commit();
+            conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcccc')");
+            conn.commit();
+            IndexRegionObserver.setSkipDataTableUpdatesForTesting(false);
+            IndexRegionObserver.setSkipPostIndexUpdatesForTesting(false);
+            selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+            // Verify that we will read from the first index table
+            assertExplainPlan(conn, selectSql, dataTableName, indexTableName + "1");
+            // Verify that one phase writes have no effect
+            rs = conn.createStatement().executeQuery(selectSql);
+            assertTrue(rs.next());
+            assertEquals("abc", rs.getString(1));
+            assertEquals("abcd", rs.getString(2));
+            assertFalse(rs.next());
+            selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val2  = 'abccc'";
+            // Verify that we will read from the second index table
+            assertExplainPlan(conn, selectSql, dataTableName, indexTableName + "2");
+            rs = conn.createStatement().executeQuery(selectSql);
+            // Verify that one phase writes have no effect
+            assertFalse(rs.next());
+            selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val2  = 'abcccc'";
+            // Verify that we will read from the second index table
+            assertExplainPlan(conn, selectSql, dataTableName, indexTableName + "2");
+            rs = conn.createStatement().executeQuery(selectSql);
+            // Verify that one phase writes have no effect
+            assertFalse(rs.next());
+        }
+    }
+
     @Test
     public void testSkipDataTableAndPostIndexPartialRowUpdate() throws Exception {
```

Comment:
Name of test is now inaccurate since we no longer skip the data table and post index write -- this appears to just be testing the happy path where an update succeeds in all 3 phases on multiple indexes. (Which is a useful test, name should just reflect it.)

Reply:
Great catch! I mistakenly removed the skip part. I will add it back.

Reply:
Done.